specification/dimensions/pricingmeasure.md

```diff
@@ -0,0 +1,33 @@
+# Pricing Measure
+
+The Pricing Measure is a provider-specified measurement unit for determining unit prices, indicating how a provider rates measured usage and purchase quantities after applying pricing rules like block pricing. Common examples include the number of hours for compute appliance runtime, gigabyte-hours for a storage appliance, or an accumulated count of requests for a network appliance or API service. Pricing Measure complements the [Pricing Quantity](#pricingquantity) measurement. Distinct from the [Usage Unit](#usageunit), it focuses on pricing and cost, not resource and service consumption, often at a coarser granularity.
+
+The PricingMeasure column MUST be present in the billing data. This column MUST be of type String. It MUST NOT contain null if PricingQuantity is not null. The PricingMeasure value MUST be aligned with the corresponding pricing measure value provided in:
```

Comment:
```suggestion
The PricingMeasure column MUST be present in the billing data. This column MUST be of type String. It MUST NOT be null if PricingQuantity is not null. The PricingMeasure value MUST be aligned with the corresponding pricing measure value provided in:
```

Reply:
Edit: 

- "be null" for consistency 
- if we impose a `MUST` condition, I believe the action should also be concrete. Align is fuzzy, hence "match" , which should be more clear as an enforceable condition.

Reply:
- Replaced `contain` with `be` -> _It MUST NOT be null if PricingQuantity is not null._
- Originally used `match` but replaced it with `be aligned with` on purpose. 
  - The reasoning behind this is that it seems those values are sometimes aligned but don't exactly match (Pricing APIs, billing data, invoices). 
  - Note that Pricing APIs and Invoices are out of our scope.

@aqu-erp Please let me know if I can close this conversation. Thanks!

Reply:
> An implementation of the [specification][] is not compliant if it fails to satisfy one or more of the "MUST"...

An alternative would be to use `SHOULD` and `match`. This way, it still prescribes "alignment" but overall spec compliance does not hinge on a fuzzy interpretation.

I'll defer to what maintainers or @udam-f2 thinks, we can resolve the conversation after.

Reply:
Here are the options:

- `MUST match`: Pricing APIs and Invoices are out of scope.
- `MUST be aligned with`: Fuzzy interpretation, verifiable by humans?
- `SHOULD match`: Compliant even if those are not aligned.
- `SHOULD match, and at least, MUST be aligned`: Sounds awkward

@udam-f2 @flanakin Need an executive decision 😄

Reply:
Hmm... I'm definitely in the "MUST" camp since "SHOULD" doesn't drive accountability. Beyond that, I'm nit-picking words. 

I suppose "be aligned with" might be subjective. "Yes, one is per 1GB and the other is 10GB, but they're both GB, so that makes them aligned." Not sure if that's a sellable argument or not 😜

"Match" is an equality statement, which is strong if there's a known text difference but if the issue is the string and not the meaning, would it make sense to say, "SHOULD be the same as and MUST semantically match"? I also thought about "quantitatively match", but I'm not sure if that's too much. There's probably a better adjective.

Reply:
> The PricingMeasure value MUST `semantically match` the corresponding pricing measure value provided in...

Probably sufficiently tackles the below scenario while retaining accountability, and sufficiently restricting definition of equivalence. 

### S3 charge, monthly invoice
| charge | price list | compliant |
-|-|-
1 GB | 1 GiB | TRUE |
1 GB | 1000 MB | FALSE
1 GB | 1024 MB | TRUE
1 GB | 953 MB | FALSE
1 GB | 1 GB-month | TRUE
1 GB | 1 GB-day | FALSE

Reply:
going with ` ...MUST be equal to the...`

@aqu-erp Please let me know if I can close this conversation. Thanks!